source/analysis/AM4DBinningAB.cpp

```diff
@@ -254,33 +256,25 @@ void AM4DBinningAB::computeCachedValues() const
 	AMnDIndex start = AMnDIndex();
 	AMnDIndex end = AMnDIndex();
 
-	switch (sumAxis_){
-
-	case 0:
-
-	    start = AMnDIndex(sumRangeMin_, 0, 0, 0);
-	    end = AMnDIndex(sumRangeMax_, inputSource_->size(1)-1, inputSource_->size(2)-1, inputSource_->size(3)-1);
-	    break;
+	if (dirtyIndices_.isEmpty()){
 
-	case 1:
-	    start = AMnDIndex(0, sumRangeMin_, 0, 0);
-	    end = AMnDIndex(inputSource_->size(0)-1, sumRangeMax_, inputSource_->size(2)-1, inputSource_->size(3)-1);
-	    break;
-
-	case 2:
-
-	    start = AMnDIndex(0, 0, sumRangeMin_, 0);
-	    end = AMnDIndex(inputSource_->size(0)-1, inputSource_->size(1)-1, sumRangeMax_, inputSource_->size(3)-1);
-	    break;
+		start = AMnDIndex(inputSource_->rank(), AMnDIndex::DoInit);
+		end = inputSource_->size()-1;
+	}
 
-	case 3:
+	else {
 
-	    start = AMnDIndex(0, 0, 0, sumRangeMin_);
-	    end = AMnDIndex(inputSource_->size(0)-1, inputSource_->size(1)-1, inputSource_->size(2)-1, sumRangeMax_);
-	    break;
+		start = dirtyIndices_.first();
+		end = dirtyIndices_.last();
 	}
```

Comment:
I notice that when each index is added to the list of dirty indices it is appended to the end, and then we assume the range here to be from .first() to .last(). Is there and case where the onInputSourceValuesChanged(...) signals might be received out of order?